source/blog/_posts/2018-01-12-mutation-testing-with-infection.md

```diff
@@ -0,0 +1,374 @@
+
+223 mutations were generated:
+     130 mutants were killed
+      63 mutants were not covered by tests
+      21 covered mutants were not detected
+       6 errors were encountered
+       3 time outs were encountered
+
+Metrics:
+         Mutation Score Indicator (MSI): 62%
+         Mutation Code Coverage: 72%
+         Covered Code MSI: 87%
+
+Please note that some mutants will inevitably be harmless (i.e. false positives).
+42.33    
+```
+
+At last we should test the threading option of inception. Be aware this option can give you many false-positives results if your tests depends on each other or use a non stateless database for testing purpose. I started infection with 4 threads and the result is amazing. Instead of 42 Seconds infection needs only 19 seconds to execute the whole mutation stuff.
+
+```bash
+    ____      ____          __  _
+   /  _/___  / __/__  _____/ /_(_)___  ____ 
+   / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
+ _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
+/___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/
+ 
+Running initial test suite...
+
+Phpunit version: 6.5.5
+
+   80 [============================] 2 secs
+
+Generate mutants...
+
+Processing source code files: 32/32
+Creating mutated files and processes: 223/223
+.: killed, M: escaped, S: uncovered, E: fatal error, T: timed out
+
+....M.M....EESM.S..............................E..   ( 50 / 223)
+................M...M.M...M...EE..E.....MMSM.M..MM   (100 / 223)
+M..SSM...........SSS..M.........SSSSSSSSSSSSSSSSSS   (150 / 223)
+SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS....S...M.....   (200 / 223)
+....M......M.M......TTT                              (223 / 223)
+
+223 mutations were generated:
+     131 mutants were killed
+      63 mutants were not covered by tests
+      20 covered mutants were not detected
+       6 errors were encountered
+       3 time outs were encountered
+
+Metrics:
+         Mutation Score Indicator (MSI): 63%
+         Mutation Code Coverage: 72%
+         Covered Code MSI: 88%
+
+Please note that some mutants will inevitably be harmless (i.e. false positives).
+19.11
+```
+
+mutation framework       | created mutations | execution time | processed mutations per second
+-------------------------|-------------------|----------------|--------------------------------
+humbug                   | 156               | 54 seconds     |  2.88 mutations
+infection                | 223               | 42 seconds     |  5.30 mutations
+infection with 4 threads | 223               | 19 seconds     | 11.74 mutations
+
+## Conclusion
+In comparison with Humbug, Infection does many things different. I like the way which infection solves the mutation challenges. I hope this tool will be still maintained in the future and we will see some new features. Maybe in the future infection didn't need to execute the whole test suite at first to do the mutation stuff. This would be another great performance improvement. :-)
```

Comment:
Compared to Humbug, Infection does a lot of things differently. I like the way, infection solves ...

Reply:
fixed